### PR TITLE
Feature/forwardpropagations

### PR DESCRIPTION
**Pull request by palmieri**
Related to the issue #8

### CODE REVIEW

File: base/PolygonMaze.h [hunk 1]
```diff
@@ -51,16 +51,17 @@ class PolygonMaze : public Environment {
...
-#endif
+        // #endif
         return true;
       }
       ++i;
     }
     return false;
   }
   bool collides(const Polygon &polygon) override {
+    std::cout << "Checking .. " << std::endl;
```
Comment (eric-heiden) [inline base/PolygonMaze.h]: maybe remove this?

File: utils/PathEvaluation.hpp [hunk 2]
```diff
@@ -36,9 +36,8 @@ struct PathEvaluation {
    * every second state.
    */
   static void computeCusps(PathStatistics &stats,
-                           const ompl::geometric::PathGeometric &p) {
+                           const std::vector<Point> path) {
```
Comment (eric-heiden) [inline utils/PathEvaluation.hpp:39]: Can you use the argument as const reference: `const std::vector<Point> &path`

**Review by eric-heiden — APPROVED** [review]
This looks good to me! Maybe you can address these small comments I added.

File: utils/PathEvaluation.hpp (hunk 2 above)
Comment (palmieri) [reply utils/PathEvaluation.hpp:39]: Fine for me.

File: base/PolygonMaze.h (hunk 1 above)
Comment (palmieri) [reply base/PolygonMaze.h]: Removed.

**Comment by roym899** [comment]
Should we merge it?

**Pull request merged**